FluidNC/src/ToolChangers/atc_manual.cpp

```diff
@@ -0,0 +1,215 @@
+        bool spindle_was_on = false;
+
+        // M6T0 is used to reset this ATC and allow us
+        if (new_tool == 0 || set_tool) {
+            _prev_tool = new_tool;
+            reset();
+            return true;
+        }
+
+        if (new_tool == _prev_tool) {
+            log_info("Requested tool already active");
+            return true;
+        }
+
+        try {
+            protocol_buffer_synchronize();  // wait for all motion to complete
+
+            if (_prev_tool == 0) {  // M6T<anything> from T0 is used for a manual change before zero'ing
+                log_info("Load first tool");
+                move_to_change_location();
+                _prev_tool = new_tool;
+                return true;
+            }
+
+            float  starting_wpos[6] = {};
+            float* mpos             = get_mpos();
+            float* wco              = get_wco();
+
+            auto n_axis = config->_axes->_numberAxis;
+            for (int idx = 0; idx < n_axis; idx++) {
+                starting_wpos[idx] = mpos[idx] - wco[idx];
+            }
+            //log_info("Starting WPos: " << starting_wpos[0] << "," << starting_wpos[1] << "," << starting_wpos[2]);
+
+            _prev_tool = new_tool;
+
+            move_to_save_z();
+            if (gc_state.modal.spindle != SpindleState::Disable) {
+                spindle_was_on = true;
+                gc_exec_linef(true, Uart0, "M5");  // this should add a delay if there is one
+            }
+
+            // if we have not determined the tool setter offset yet, we need to do that.
+            if (!_have_tool_setter_offset) {
+                move_over_toolsetter();
+                // do a seek probe if needed
+                if (!seek_probe())
+                    return false;
+                // do the position finding feed rate probe
+                if (!probe(_probe_feed_rate, _tool_setter_position)) {
+                    return false;
+                }
+
+                _tool_setter_offset      = _tool_setter_position[2];
+                _have_tool_setter_offset = true;
+            }
+
+            move_to_change_location();
+            log_info("Please install tool:" << new_tool);
+
+            if (!hold_and_wait_for_resume()) {
+                log_info("Tool change aborted");
+                return false;  // aborted
+            }
+            // probe the new tool
+            move_to_save_z();
+            move_over_toolsetter();
+
+            if (!seek_probe())
+                return false;
+
+            float offset_probe[MAX_N_AXIS] = {};
+            if (!probe(_probe_feed_rate, offset_probe)) {
+                return false;
+            }
+
+            float tlo = offset_probe[2] - _tool_setter_offset;
+
+            //log_info("Set TLO:" << tlo);
+            gc_exec_linef(false, Uart0, "G43.1Z%0.3f", tlo);
+
+            move_to_save_z();
+            gc_exec_linef(false, Uart0, "G0X%0.3fY%0.3f", starting_wpos[0], starting_wpos[1]);
```

Comment:
I worry that these gc_exec_linef()s could block waiting for the planner, possibly resulting in deadlocks